README.md

```diff
@@ -9,3 +9,664 @@ ExpressRouterAdapter is available as a package on NPM for use with a module bund
+        .allowAnonymous()
+        .put(({ pedId }) => {
+          // return nothing
+        })
+    ];
+  }
+}
+```
+
+Then you will automatically get a 204.
+
+
+## What if it does return something?
+
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(({ petId }) => {
+          return { petId, name: 'honey' }
+        })
+    ];
+  }
+}
+```
+
+
+Then you will automatically get a 200 with your model serialized as JSON and a content-type header of application/json.
+
+## What if I need do something asynchronous
+
+```js
+
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          await doSomethingAsync(petId);
+        })
+    ];
+  }
+}
+```
+
+
+
+## What if I throw an exception
+```js
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          throw new Error(`there be gremlins`);
+        })
+    ];
+  }
+}
+```
+then it will bubble to the express middleware to your error handler. If you haven't sent one up, here is an example.
+
+```js
+app.use(err: any, req: any, res: any, next: any): void {
+  const {
+    status = 500,
+    message = 'Server Error'
+  } = err;
+  if (status >= 500) {
+    console.log('failed to process request', req.method, req.path);
+    console.error('error handler error', err);
+  }
+
+  let body = err.body || {
+      status,
+      message
+  };
+
+  res.status(status)
+    .json(body);
+}
+
+```
+
+## What if I want more control over the status code or headers
+ ```js
+ import { HTTPResponse, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          return new HTTPResponse({
+            status: 400,
+            body: {
+              message: `pet was missing name`,
+              code: 123
+            }
+          });
+        })
+    ];
+  }
+}
+```
+
+You also can throw
+```js
+ import { HTTPError, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
+class PetRouter{
+  getRoutes(){
+    return [
+      new RouterMetaBuilder()
+        .path('/pets/:petId')
+        .allowAnonymous()
+        .put(async ({ petId }) => {
+          throw new HTTPError({
+            message: `pet validation failed`,
+            status: 400,
+            body: {
+              message: `pet was missing name`,
+              code: 123
+            }
+          });
+        })
+    ];
+  }
+}
+```
+
+Which is convenient as you abstract things away, so you can end up with code like
+```js
+ import { HTTPError, RouterMetaBuilder } from '@symbiotic/express-router-adapter';
```

Comment:
We aren't using HTTPError that we are importing

Should consider adding a comment to validataeCreatePet method saying it throws an HTTPError

Reply:
Remove import, add comment that validateCreatePet throws an HTTPError